src/vm/types.rs

```diff
@@ -41,6 +41,14 @@ pub struct TypeSignature {
     //       high dimensional lists are _expensive_ --- use lists of tuples!
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct FunctionArg {
```

Comment:
What happens if I use the same state db with a non-developer-mode build?  Or, what happens if I use a non-developer-mode db with a developer-build binary?  Do things break for want of the missing (or unexpectedly-present) `name` field?

Reply:
Yes -- things would break here. I think the best course of action here is to use check_schema to panic when the supplied schema is inappropriate. I think it would be too much unneeded complexity to try and make it so that the schema is compatible between the two builds.

Reply:
dev-mode -> prod-mod db would be fine since we aren't using `#[serde(deny_unknown_fields)]`.

prod-mode -> dev-mode db would need `#[serde(skip_deserializing)]` on the `name` field. 

A check_schema or similar db mode/version flag check could work. 

What about allowing the `name` field to be serialized without a dev-mode exclusion? Could be useful for improving error messages -- in general, include in prod. 

I have no strong opinion either way.

Reply:
I'd prefer the `check_schema` approach.  Then we don't have to worry about converting to/from dev/prod modes at all.  If we later feel the need to do the conversion, we can have a separate command for doing so.

Reply:
I agree about not doing conversions. What about removing the `develop-mode` exclusion for the `name` field? 
> Could be useful for improving error messages -- in general, include in prod.